### ISSUE TITLE
cop for enforcing `should` vs. `expect` with explicit subject

### ISSUE THREAD

**Issue by sauloperez** [rubocop-rspec/rubocop-rspec#587]
At first, I thought https://github.com/rubocop-rspec/rubocop-rspec/pull/200 also covers the case where are an explicit subject is used such as the one below:

```ruby
  it "computes shipping cost for a line item" do
    variant = double(:variant, :weight => 10)

    line_item = double(:line_item, :variant => variant, :quantity => 2)

    subject.set_preference(:per_kg, 10)
    subject.compute(line_item).should == 10*2 * 10
  end
```

then I realized that it is only for implicit subjects. Is there any cop that works in this case too (couldn't find any)? we have quite a big codebase and we want to discourage `should` syntax.

**Comment by bquorning** [comment]
Do I understand correctly that you would like to enforce using 
```rb
expect(subject.compute(line_item)).to eq(10*2 * 10)
```
instead of 
```rb
subject.compute(line_item).should == 10*2 * 10
```

That is a feature of RSpec itself – here you can read about how to enable/disable one or the other: https://relishapp.com/rspec/rspec-expectations/docs/syntax-configuration#disable-should-syntax

I personally prefer to configure RSpec to not do any monkey patching at all (no `should` method on every object, and the outer-most `describe` or `context` must be called directly on `RSpec`): https://relishapp.com/rspec/rspec-core/v/3-7/docs/configuration/zero-monkey-patching-mode#monkey-patched-methods-are-undefined-with-%60disable-monkey-patching!%60

**Comment by sauloperez** [comment]
Yep, that's what I meant. I thought I could postpone disabling the syntax and giving Transpec a try by at least stopping new additions of `should`.

Thanks for the help :ok_hand:

**Comment by sauloperez** [comment]
That would be helpful and I guess it will reduce the number of issues you need to answer.

**Comment by bquorning** [comment]
That sounds like a good pull request 😄

**Comment by sauloperez** [comment]
I can give it a try if you want

**Comment by bquorning** [comment]
Fixed by #588.
